src/main/java/cceclipseplugin/core/PluginManager.java

```diff
@@ -184,7 +184,10 @@ private void registerWSHooks() {
 			String password = prefStore.getString(PreferenceConstants.PASSWORD);
 			boolean showWelcomeDialog = (username == null || username.equals("") || password == null || password.equals(""));
 			if (showWelcomeDialog) {
-				Display.getDefault().asyncExec(() -> new WelcomeDialog(new Shell(), prefStore).open());
+				Display.getDefault().asyncExec(() -> {
+                    Shell shell = Display.getDefault().getActiveShell();
+				    new WelcomeDialog(shell, prefStore).open();
+                });
```

Comment:
This indentation looks weird. It might be my pull request UI glitching, but it looks like lines 190 and 188 are too far left (like, further left than `if (showWelcomeDialog)` ).